registry/cache/cache.go

```diff
@@ -101,38 +100,44 @@ func (c *cache) quit() bool {
 	}
 }
 
-func (c *cache) del(service string) {
+func (c *cache) del(domain, service string) {
 	// don't blow away cache in error state
 	if err := c.status; err != nil {
 		return
 	}
-	// otherwise delete entries
-	delete(c.cache, service)
-	delete(c.ttls, service)
+
+	if _, ok := c.services[domain]; ok {
+		delete(c.services[domain], service)
+	}
+
+	if _, ok := c.ttls[domain]; ok {
+		delete(c.ttls, service)
```

Comment:
Should this be `delete(c.ttls[domain], service)` ?

Reply:
Yes, good catch.